### ISSUE TITLE
configure not honouring OpenSSL options properly

### ISSUE THREAD

**Issue by clinew** [atheme/libmowgli-2#42]
```
Script started on Fri Sep 15 12:44:02 2017
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m ./configure | grep OpenSSL
checking whether compiling and linking against 01;31mKOpenSSLmK works... yes
   01;31mKOpenSSLmK support: yes
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m K./configure 1@1@-1@-1@w1@i1@t1@h1@o1@u1@t1@-1@o1@p1@e1@n1@s1@s1@11111111111P1@-1@1P1@o1@p1@e1@n1@s1@s1@l1@1P
checking whether compiling and linking against 01;31mKOpenSSLmK works... no
   01;31mKOpenSSLmK support: no
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m K./configure --with-openssl1@=1@y1@e1@sSSL
checking whether compiling and linking against 01;31mKOpenSSLmK works... no
   01;31mKOpenSSLmK support: no
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m K./configure --with-openssl=ye111P1@n1@openSSL
   01;31mKOpenSSLmK support: no
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m K./configure 17P1@-1@-1@w1@i1@t1@h1@o1@u1@t1@-1@o1@p1@e1@n1@s1@s1@l
   01;31mKOpenSSLmK support: no
01;32mfrostsnow@test01;34m ~/libmowgli-2 $00m Kexit

Script done on Fri Sep 15 12:45:30 2017
```
Ugh, why is 'col' output so awful?

In summary:
Option | Got | Expected
no options | Y | Either
--with-openssl | N | Y
--with-openssl=yes | N | Y
--with-openssl=no | N | N
--without-openssl | N | N

**Comment by grawity** [comment]
--with-openssl=DIR      root of the OpenSSL directory

It's not a boolean; you're trying to give it `yes` and `no` as filesystem paths.

**Comment by aaronmdjones** [comment]
Should be resolved by commit 2f8f6a3e39777f31

**Comment by clinew** [comment]
Works for me, thank you!
